feat(array-methods-2): animal object challenge (#97)
neuefische/web-exercises: sessions/js-array-methods-2/animal-objects/README.md

# Array Methods 2: An Array with Objects

> This codesandbox offers a some exercises to recap some enhanced array methods. Let's dive directly into it!

Reviewer: ```suggestion
This CodeSandbox offers a some more exercises to recap some advanced array methods. Let's dive in!
```

## Task

In the [index.js](index.js) file, there is an `animals` array with - guess what? - animals. Each animal is an object with some key-value-pairs.

Below this array, you will find a couple of variables (e.g. `firstAnimalStartingWithLetterG`). They all are initialized with `null`.

> And now it's your turn: delete the value `null` and write code such that the variables have a value corresponding to their name. Use the following methods to achieve this goal:
> 
> - `Array.includes`()`
> - `Array.find`()`
> - `Array.findIndex`()`
> - `Array.sort`()`
> - `Array.some`()`
> - `Array.every`()`
> - `Array.reduce`()`
> - `Array.reverse`()`
> 
> Additionally, at some point you will need:
> 
> - `Array.filter`()`
> - `Array.map`()`
> - `String.startsWith`()`

Reviewer 2: ```suggestion
And now it's your turn: delete the value `null` and write code such that the variables have a value corresponding to their name. Use the following methods to achieve this goal:

- `Array.includes()`
- `Array.find()`
- `Array.findIndex()`
- `Array.sort()`
- `Array.some()`
- `Array.every()`
- `Array.reduce()`
- `Array.reverse()`

Additionally, at some point you will need:

- `Array.filter()`
- `Array.map()`
- `String.startsWith()`
```
